src/transformers/models/iris/modeling_iris.py

```diff
@@ -0,0 +1,1964 @@
+            down.attn = attn
+            if i_level != self.num_resolutions - 1:
+                down.downsample = Downsample(block_in, with_conv=True)
+                curr_res = curr_res // 2
+            self.down.append(down)
+
+        # middle
+        self.mid = nn.Module()
+        self.mid.block_1 = ResnetBlock(
+            in_channels=block_in, out_channels=block_in, temb_channels=temb_ch, dropout=config.dropout
+        )
+        self.mid.attn_1 = AttnBlock(block_in)
+        self.mid.block_2 = ResnetBlock(
+            in_channels=block_in, out_channels=block_in, temb_channels=temb_ch, dropout=config.dropout
+        )
+
+        # end
+        self.norm_out = Normalize(block_in)
+        self.conv_out = torch.nn.Conv2d(block_in, config.z_channels, kernel_size=3, stride=1, padding=1)
+
+    def forward(
+        self, x: torch.Tensor, output_hidden_states: bool = False, output_attentions: bool = False
+    ) -> tuple[torch.Tensor, list[torch.Tensor]]:
+        temb = None  # timestep embedding
+
+        # downsampling
+        hs = [self.conv_in(x)]
+
+        hidden_states = () if output_hidden_states else None
+        attentions = () if output_attentions else None
+
+        for i_level in range(self.num_resolutions):
+            for i_block in range(self.config.num_res_blocks):
+                h = self.down[i_level].block[i_block](hs[-1], temb)
+                if len(self.down[i_level].attn) > 0:
+                    attn_out = self.down[i_level].attn[i_block](h)
+                    h = attn_out[0]
+                    # Add attention weights
+                    attentions = attentions + (attn_out[1],) if output_attentions else None
+                hs.append(h)
+                # Add the hidden state to the tuple
+                hidden_states = hidden_states + (h,) if output_hidden_states else None
+            if i_level != self.num_resolutions - 1:
+                hs.append(self.down[i_level].downsample(hs[-1]))
+
+        # middle
+        h = hs[-1]
+        h = self.mid.block_1(h, temb)
+        # Add the hidden state to the tuple
+        hidden_states = hidden_states + (h,) if output_hidden_states else None
+        attn_out = self.mid.attn_1(h)
+        h = attn_out[0]
+        # Add attention weights
+        attentions = attentions + (attn_out[1],) if output_attentions else None
+        # Add the hidden state to the tuple
+        hidden_states = hidden_states + (h,) if output_hidden_states else None
+        h = self.mid.block_2(h, temb)
+        # Add the hidden state to the tuple
+        hidden_states = hidden_states + (h,) if output_hidden_states else None
+
+        # end
+        h = self.norm_out(h)
+        h = nonlinearity(h)
+        h = self.conv_out(h)
+
+        return h, hidden_states, attentions
+
+
+class Decoder(nn.Module):
+    def __init__(self, config: EncoderDecoderConfig) -> None:
+        super().__init__()
+        self.config = config
+        temb_ch = 0
```

Comment:
Could we use a more descriptive name here ?

Reply:
yeah, these are token embedding dimensions, will change the var name :+1: